pxr/usd/sdf/testenv/testSdfPath.py

```diff
@@ -140,17 +157,80 @@ def BasicTest(path, elements):
 
             # Test path
             testPaths.append(Sdf.Path(string))
-            BasicTest(testPaths[-1], testElements[testIndex])
+            BasicTest(testPaths[-1], testElements[testIndex], testAbsolute, 
+                testProperty, testIndex)
         
             # If path is a property then try it with a namespaced name.
             if testProperty[testIndex]:
                 testPaths.append(Sdf.Path(string + ':this:has:namespaces'))
                 elements = list(testElements[testIndex])
                 elements[-1] += ':this:has:namespaces'
-                BasicTest(testPaths[-1], elements)
+                BasicTest(testPaths[-1], elements, testAbsolute,
+                    testProperty, testIndex)
         
         print('\tPassed')
         
+        # ========================================================================
+        # Test UTF-8 encoded paths
+        # ========================================================================
+        utf8_good_paths = [
+            "123",
+            "123test",
```

Comment:
The Nd category (decimal digits) is not part of xid_start, so these paths should _not_ be considered good, correct?  Or am I missing something here?

Reply:
I think it is ok if non-ASCII digits sort incorrectly. However it is pretty important that numbers sort as before, and also that punctuation effect the sort exactly as though the words separated by them were individually sorted. USD relies on this, putting '/' and '.' into strings and expecting them to sort in the same order as the same strings truncated at those punctuation marks.
I think a method that would work is to seqment the strings being compared into alternating sequences of non-alphabetic ASCII (N) and all other characters (all UTF-8 and all ASCII letters) (L). (N) strings are compared using the normal sorting (which puts numbers in order). (L) strings are compared using UTF-8 collating (putting words in alphabetical order and mixing case). A string that starts with (N) is strictly less than a string that starts with (L).